File: 兴趣班入门百练讲义/NQ100/NQ055/NQ055_improved.cpp
#include <iostream>
using namespace std;

// 求最大公约数（GCD）
int gcd(int a, int b) {
    return b == 0 ? a : gcd(b, a % b);
}

// 求最小公倍数（LCM）
int lcm(int a, int b) {
    return (a * b) / gcd(a, b);
}

int main() {
    int n;
    while (cin >> n && n > 0) {
        int result = 1;
        for (int i = 0; i < n; i++) {
            int num;
            cin >> num;
            result = lcm(result, num); // 累积计算LCM
        }
        cout << result << endl;
    }
    return 0;
}

File: 兴趣班入门百练讲义/NQ100/NQ055/NQ055_improved.py
import sys

def gcd(a, b):
    """求最大公约数（GCD）"""
    while b:
        a, b = b, a % b
    return a

def lcm(a, b):
    """求最小公倍数（LCM）"""
    return (a * b) // gcd(a, b)

def main():
    input = sys.stdin.read().split()
    ptr = 0
    
    while ptr < len(input):
        n = int(input[ptr])
        ptr += 1
        if n <= 0:
            break
        
        result = 1
        for _ in range(n):
            num = int(input[ptr])
            ptr += 1
            result = lcm(result, num)  # 累积计算LCM
        
        print(result)

if __name__ == "__main__":
    main()